File: airbyte-connector-builder-server/src/main/java/io/airbyte/connector_builder/exceptions/CdkProcessExceptionHandler.java
/*
 * Copyright (c) 2020-2025 Airbyte, Inc., all rights reserved.
 */

package io.airbyte.connector_builder.exceptions;

import io.micronaut.context.annotation.Requires;
import io.micronaut.http.HttpRequest;
import io.micronaut.http.HttpResponse;
import io.micronaut.http.annotation.Produces;
import io.micronaut.http.server.exceptions.ExceptionHandler;
import jakarta.inject.Singleton;

/**
 * Custom Micronaut exception handler for the {@link CdkProcessException}.
 */
@Produces
@Singleton
@Requires(classes = CdkProcessException.class)
public class CdkProcessExceptionHandler implements ExceptionHandler<CdkProcessException, HttpResponse> {

  final ExceptionHelper helper = new ExceptionHelper();

  @Override
  public HttpResponse handle(final HttpRequest request, final CdkProcessException exception) {
    return helper.handle(request, exception);
  }

}

File: airbyte-connector-builder-server/src/main/kotlin/io/airbyte/connectorbuilder/exceptions/CdkProcessExceptionHandler.kt
/*
 * Copyright (c) 2020-2025 Airbyte, Inc., all rights reserved.
 */

package io.airbyte.connectorbuilder.exceptions

import io.micronaut.context.annotation.Requires
import io.micronaut.http.HttpRequest
import io.micronaut.http.HttpResponse
import io.micronaut.http.annotation.Produces
import io.micronaut.http.server.exceptions.ExceptionHandler
import jakarta.inject.Singleton

/**
 * Custom Micronaut exception handler for the [CdkProcessException].
 */
@Produces
@Singleton
@Requires(classes = [CdkProcessException::class])
class CdkProcessExceptionHandler : ExceptionHandler<CdkProcessException, HttpResponse<*>> {
  val helper: ExceptionHelper = ExceptionHelper()

  override fun handle(
    request: HttpRequest<*>,
    exception: CdkProcessException,
  ): HttpResponse<*> = helper.handle(request, exception)
}
